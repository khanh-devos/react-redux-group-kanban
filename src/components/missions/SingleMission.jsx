import { PropTypes } from 'prop-types';
import { useDispatch } from 'react-redux';
import { toggleJoinMission } from '../../redux/missions/missionsSlice';

export default function SingleMission({
  missionId, missionName, missionDescription, missionReserved,
}) {
  const dispatch = useDispatch();
  const handleJoinMission = (e) => {
    const missionId = e.target.id;
    dispatch(toggleJoinMission(missionId));
  };
  return (
    <tr>
      <td>
        {missionName}
      </td>
      <td>
        {missionDescription}
      </td>
      <td>
        { missionReserved && (
          <button
            type="button"
            className="mission-active-button"
          >
            Active Member
          </button>
        )}
        { !missionReserved && (
          <button
            type="button"
            className="mission-inactive-button"
          >
            NOT A MEMBER
          </button>
        )}
      </td>
      <td>
        { missionReserved && (
          <button
            id={missionId}
            type="button"
            onClick={handleJoinMission}
            className="leave-mission-button"
          >
            Leave Mission
          </button>
        )}
        { !missionReserved && (
          <button
            id={missionId}
            type="button"
            title="test-mission-btn"
            onClick={handleJoinMission}
            className="join-mission-button"
          >
            Join Mission
          </button>
        )}
      </td>
    </tr>
  );
}

SingleMission.propTypes = {
  missionId: PropTypes.string.isRequired,
  missionName: PropTypes.string.isRequired,
  missionDescription: PropTypes.string.isRequired,
  missionReserved: PropTypes.bool.isRequired,
};
