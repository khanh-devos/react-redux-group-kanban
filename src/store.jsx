import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './redux/missions/missionsSlice';
import rocketReducer from './redux/rocketSlice/rocketSlice';

const store = configureStore({
  reducer: {
    rocket: rocketReducer,
    mission: missionsReducer,
  },
});

export default store;
