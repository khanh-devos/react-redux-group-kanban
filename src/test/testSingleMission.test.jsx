import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import SingleMission from '../components/missions/SingleMission';
import store from '../store';

describe('test Single Mission', () => {
  it('should match Single Mission', () => {
    const { container } = render(
      <Provider store={store}>
        <SingleMission missionId="111" missionName="name" missionDescription="desc" missionReserved={false} />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
