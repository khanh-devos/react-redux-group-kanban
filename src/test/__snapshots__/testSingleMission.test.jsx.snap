// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Single Mission should match Single Mission 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <td>
          name
        </td>
        <td>
          desc
        </td>
        <td>
          <button
            class="mission-inactive-button"
            type="button"
          >
            NOT A MEMBER
          </button>
        </td>
        <td>
          <button
            class="join-mission-button"
            id="111"
            title="test-mission-btn"
            type="button"
          >
            Join Mission
          </button>
        </td>
      </tr>
    </thead>
  </table>
</div>
`;
